firmware/library/L1_Drivers/ssp.hpp

```diff
@@ -61,12 +63,227 @@ class SspInterface
 
     virtual bool Initialize()                                       = 0;
     virtual bool GetTransferStatus()                                = 0;
-    virtual uint8_t Transfer(uint16_t data)                         = 0;
+    virtual uint16_t Transfer(uint16_t data)                        = 0;
     virtual void SetPeripheralPower()                               = 0;
     virtual void SetPeripheralMode(
             MasterSlaveMode mode, FrameMode frame, DataSize size)   = 0;
-    // NOTE: clock_divider is the number of prescaler-output clocks per bit.
+    // NOTE: "divider" is the number of prescaler-output clocks per bit.
     // See page 611 of UM10562 for SCR-Serial Clock Rate for more info.
     virtual void SetClock(bool polarity, bool phase,
-            uint8_t prescaler, uint8_t clock_divider)               = 0;
+            uint8_t prescaler, uint8_t divider)                     = 0;
+};
+
+class Ssp : public SspInterface
+{
+ public:
+    // SSP register lookup table
+    static LPC_SSP_TypeDef * ssp_registers[2];
+    static constexpr uint8_t kEnableSspPins = 0b010;
+    static constexpr uint8_t kPort0 = 0;
+    static constexpr uint8_t kPin18 = 18;
+    static constexpr uint8_t kPin17 = 17;
+    static constexpr uint8_t kPin15 = 15;
+    static constexpr uint8_t kPin9 = 9;
+    static constexpr uint8_t kPin8 = 8;
+    static constexpr uint8_t kPin7 = 7;
+    static constexpr uint8_t kDataBit = 0;
+    static constexpr uint8_t kFrameBit = 4;
+    static constexpr uint8_t kPolarityBit = 6;
+    static constexpr uint8_t kPhaseBit = 7;
+    static constexpr uint8_t kDividerBit = 8;
+    static constexpr uint8_t kMasterModeBit = 2;
+    static constexpr uint8_t kDataLineIdleBit = 4;

```

Comment:
Nope, because they are super specific to this exact implementation. The interface should be usable across MCUs, CPUs ect. We should be able to swap out your SPI implementation with one meant for the PIC and it should still be able to properly reference the SpiInterface.